### PR TITLE
Add API calls of Windows Heap for Cutter

### PR DESCRIPTION
**Pull request by MalhotraPulak**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

Added two API calls, one gets the data related to `dmwb` command and one for `dmw` command. 
Will mark this ready for review once the Cutter part looks satisfactory. 

**Test plan**

Use the Cutter Widget and see if it works fine

**Closing issues**

...

### CODE REVIEW

File: librz/core/cmd_windows_heap.c [hunk 1]
```diff
@@ -35,4 +35,24 @@ RZ_IPI RzCmdStatus rz_cmd_debug_process_heaps_handler(RzCore *core, int argc, co
 RZ_IPI RzCmdStatus rz_cmd_debug_process_heap_block_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
 	NOT_SUPPORTED_ERROR_MESSAGE;
 }
+#endif
+
+/* API calls for Cutter */
+
```
Comment (XVilka) [inline librz/core/cmd_windows_heap.c]: Since those aren't commands, what about moving them in a separate file? `librz/core/cheap.c` or something.

Comment (MalhotraPulak) [reply librz/core/cmd_windows_heap.c]: sure I can do that

**Comment by XVilka** [comment]
Please rebase on top of the latest `dev`

File: librz/core/ctypes.c [hunk 2]
```diff
@@ -305,20 +305,23 @@ static void core_types_struct_print(RzCore *core, RzBaseType *btype, RzOutputMod
 	switch (mode) {
 	case RZ_OUTPUT_MODE_JSON: {
 		rz_return_if_fail(pj);
+		if (!btype) {
+			break;
+		}
```
Comment (ret2libc) [inline librz/core/ctypes.c]:
```suggestion
```

Is this the same as the previous PR or another case?
Anyway, this is not necessary.

File: librz/core/ctypes.c [hunk 3]
```diff
@@ -305,20 +305,23 @@ static void core_types_struct_print(RzCore *core, RzBaseType *btype, RzOutputMod
...
 		pj_o(pj);
-		if (btype && !rz_vector_empty(&btype->struct_data.members)) {
-			pj_ks(pj, "name", btype->name);
-			pj_k(pj, "members");
-			pj_o(pj);
+		pj_ks(pj, "name", btype->name);
+		pj_k(pj, "members");
+		pj_o(pj);
+		if (!rz_vector_empty(&btype->struct_data.members)) {
```
Comment (ret2libc) [inline librz/core/ctypes.c]: remove the `if` completely, as in the other PR.

File: librz/core/ctypes.c (hunk 2 above)
Comment (XVilka) [reply librz/core/ctypes.c]: This commit just could be dropped from the PR, it's easier.

File: librz/core/cheap.c [hunk 4]
```diff
@@ -0,0 +1,23 @@
+#include <rz_core.h>
+#include "core_private.h"
+
+/* API calls of windows heap for Cutter */
+#if __WINDOWS__
+RZ_API RzList *rz_heap_windows_blocks_list(RzCore *core) {
```
Comment (XVilka) [inline librz/core/cheap.c]: Please add if the list owned or borrowed by the caller: https://github.com/rizinorg/rizin/blob/dev/DEVELOPERS.md#custom-pointer-modifiers

File: librz/core/cheap.c [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+	return rz_heap_blocks_list(core);
+}
+
+RZ_API RzList *rz_heap_windows_heap_list(RzCore *core) {
```
Comment (XVilka) [inline librz/core/cheap.c]: Same here (also duplicate those in the header files)

**Review by XVilka — CHANGES_REQUESTED** [review]

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**